nonce.go

```diff
@@ -63,6 +62,26 @@ func (ns Nonces) Validate() error {
 	return nil
 }
 
+// Append a nonce to a nonces' list
+func (ns *Nonces) Append(n Nonce) {
+	*ns = append([]Nonce(*ns), n)
+}
+
+// Len returns the number of nonces in a list
+func (ns Nonces) Len() int {
+	return len(ns)
+}
+
+// GetI returns the nonce found at index (starting at 0) or nil if the index is
+// out of bounds
+func (ns Nonces) GetI(index int) []byte {
```

Comment:
The interface defined at Line 77 is quite the opposite of the general philosophy of checking nonce validity (example line 102). Is it possible to return an error instead of nil here if index is out of bounds?

Reply:
I wanted to make this coherent with the `Get()` interface (L125).